phpunit/html/wp-html-tag-processor-test.php

```diff
@@ -151,6 +151,18 @@ public function test_attributes_parser_treats_slash_as_attribute_separator() {
 		$this->assertSame( 'test', $p->get_attribute( 'e' ), 'Accessing an existing e="test" did not return "test"' );
 	}
 
+	/**
+	 * @covers WP_HTML_Tag_Processor::get_attribute_names
+	 */
+	public function test_get_attribute_names() {
+		$p = new WP_HTML_Tag_Processor( '<div enabled class="test" data-test-id="14">Test</div>' );
```

Comment:
Let's also test what happens when:

* There are no attributes on a given tag
* An attribute was added but not flushed
* An attribute was added and flushed

Any other corner cases you can think of @dmsnell ?

Reply:
Thank you, these are all great suggestions -- I'll add them 👍 

(Even if we [proceed with `get_attribute_names_prefixed_by()`](https://github.com/WordPress/gutenberg/pull/46685#issuecomment-1368955554), we should be able to carry them over 😊 )

Reply:
I've added tests for the following in ca06c6483a8a9399f401f5bf5a35bc4c463e2382:

> * There are no attributes on a given tag
> * We're on a tag closer
> * We're not on any tag (before the first `next_tag`)

(Mostly by copying and modifying the corresponding `get_attribute()` tests.) This also included a test that navigates to a non-existing tag and verifies that `get_attribute_names()` returns `null`.)

Since there wasn't a test to check that `get_attribute()` returned `null` when in a closing tag, I've added one in 8c51859ddc4eadb1ae483cd692f0b41f966247c6.

Reply:
As for the remaining test cases

> An attribute was added but not flushed
> An attribute was added and flushed

...I'd like to wait for the outcome of #46680 before adding coverage for "added but not flushed", since we might need it.

"Added and flushed" I'll add now 👍

Reply:
> "Added and flushed" I'll add now 👍

5b248c0